consul/base.py

```diff
@@ -1910,20 +1949,24 @@ def info(self,
                 '/v1/session/info/%s' % session_id,
                 params=params)
 
-        def renew(self, session_id, dc=None):
+        def renew(self, session_id, dc=None, token=None):
             """
             This is used with sessions that have a TTL, and it extends the
             expiration by the TTL.
 
             *dc* is the optional datacenter that you wish to communicate with.
             If None is provided, defaults to the agent's datacenter.
 
+            *token* is an optional `ACL token`_ to apply to this request.
+
             Returns the session.
             """
             params = []
             dc = dc or self.agent.dc
             if dc:
                 params.append(('dc', dc))
+            if token:
```

Comment:
```suggestion
            token = token or self.agent.token
            if token:
```
agent may already have token set. Use it!